internal/installation/paths.go

```diff
@@ -66,20 +60,20 @@ func BinPathFromInstallPath(installPath string) (string, error) {
 	return filepath.Join(installPath, BinDirName), nil
 }
 
-func InstalledOnPath(installPath string) (bool, string, error) {
-	path := appinfo.StateApp(installPath).Exec()
-	exists := fileutils.TargetExists(path)
-	if exists {
-		return exists, installPath, nil
-	}
-
-	binPath, err := BinPathFromInstallPath(installPath)
-	if err != nil {
-		return false, "", errs.Wrap(err, "Could not detect binPath from BinPathFromInstallPath")
+func InstalledOnPath(installRoot string) (bool, string, error) {
+	var binPath string
+	var err error
+	if filepath.Base(installRoot) == "bin" {
+		binPath = installRoot
```

Comment:
I think this is where we can use the InstallDirMarker. Rather than make assumptions about file paths we can just scan up the tree until we find InstallDirMarker.